power_grid_model_c/power_grid_model/include/power_grid_model/math_solver/common_solver_functions.hpp

```diff
@@ -57,17 +57,91 @@ template <symmetry_tag sym> inline void copy_y_bus(YBus<sym> const& y_bus, Compl
     });
 }
 
-template <symmetry_tag sym>
-inline void calculate_source_result(IdxRange const& sources, Idx bus_number, YBus<sym> const& y_bus,
-                                    PowerFlowInput<sym> const& input, SolverOutput<sym>& output) {
+// Calculates current and power injection of source i for multiple symmetric sources at a node.
+// The current injection of source i to the bus in phase space is:
+//     i_inj_i = (y_ref_i * z_ref_t) [ (u_ref_i * y_ref_t - i_ref_t) + i_inj_t]
+// Phase space is used because y_ref is a DoubleComplex in the symmetric case.
+// The equation is organized in such a way to avoid the numerical instability induced in the substraction if y_ref_i
+// is large.
```

Comment:
```suggestion
/// @brief Calculates current and power injection of source i for multiple symmetric sources at a node.
/// The current injection of source i to the bus in phase space is:
///     i_inj_i = (y_ref_i * z_ref_t) [ (u_ref_i * y_ref_t - i_ref_t) + i_inj_t]
/// The equation is organized in such a way to avoid the numerical instability induced in the substraction if y_ref_i
/// is large.
```
- docstring would be recommended instead of normal comments.
- Also, there are no phases in symmetrical case in a mathematical sense.